plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/WSTUtils.java

```diff
@@ -186,11 +190,33 @@ public static IServer extractServer(final String browserLocation, final List<ISe
 	}
 
 	/**
+	 * Retrieves the hostname for the given server, with support for specific types of servers, such as OpenShift 3.
+	 * @param server the server to analyze
+	 * @return the server hostname
+	 * @throws ExecutionException if something went wrong
+	 */
+	public static String getWebHost(final IServer server) throws ExecutionException {
+		final String serverType = server.getServerType().getId();
+		final Object adapter = server.loadAdapter(ServerDelegate.class, null);
+		if (serverType.equals(OPENSHIFT_SERVER_TYPE)) {
+			if(adapter instanceof IExtendedPropertiesProvider) {
+				final IExtendedPropertiesProvider propertiesProvider = (IExtendedPropertiesProvider) adapter;
+				try {
+					final URL welcomePageURL = new URL(propertiesProvider.getExtendedProperties().getWelcomePageUrl());
+					return welcomePageURL.getHost();
+				} catch (MalformedURLException | GetWelcomePageURLException e) {
+					throw new ExecutionException("Failed to retrieve the hostname for the selected server", e );
```

Comment:
Could there be a way for the case of GetWelcomePageURLException to show a message dialog informing user of the problem? This exception is thrown exactly for that purpose and is shown as a dialog in Show in Browser option, when there is a problem with the connection or service that is not a coding part on our part but something user should fix in his settings, in such cases error in a log usually is not a user friendly way to inform him of _his_ problems.